File: modules/org.openbravo.retail.posterminal/src/org/openbravo/retail/posterminal/master/BPCategory.java
/*
 ************************************************************************************
 * Copyright (C) 2012 Openbravo S.L.U.
 * Licensed under the Openbravo Commercial License version 1.0
 * You may obtain a copy of the License at http://www.openbravo.com/legal/obcl.html
 * or in the legal folder of this module distribution.
 ************************************************************************************
 */
package org.openbravo.retail.posterminal.master;

import java.util.ArrayList;
import java.util.List;

import org.codehaus.jettison.json.JSONException;
import org.codehaus.jettison.json.JSONObject;
import org.openbravo.retail.posterminal.ProcessHQLQuery;

public class BPCategory extends ProcessHQLQuery {

  @Override
  protected List<String> getQuery(JSONObject jsonsent) throws JSONException {

    List<String> hqlQueries = new ArrayList<String>();

    // standard product categories
    hqlQueries
        .add("select bpcat.id as id, bpcat.searchKey as searchKey,bpcat.name as name, bpcat.name as _identifier "
            + "from BusinessPartnerCategory bpcat "
            + "where "
            + "(bpcat.$incrementalUpdateCriteria) AND (bpcat.$incrementalUpdateCriteria)) order by bpcat.name");

    return hqlQueries;
  }
}

File: ${env.CATALINA_HOME}/webapps/openbravo/web/org.openbravo.retail.posterminal/js/model/bpcategory.js
/*
 ************************************************************************************
 * Copyright (C) 2012 Openbravo S.L.U.
 * Licensed under the Openbravo Commercial License version 1.0
 * You may obtain a copy of the License at http://www.openbravo.com/legal/obcl.html
 * or in the legal folder of this module distribution.
 ************************************************************************************
 */

/*global Backbone */

(function () {

  var BPCategory = Backbone.Model.extend({
    modelName: 'BPCategory',
    tableName: 'c_bp_group',
    entityName: 'BPCategory',
    source: 'org.openbravo.retail.posterminal.master.BPCategory',
    dataLimit: 300,
    properties: ['id', 'searchKey', 'name', '_identifier', '_idx'],
    propertyMap: {
      'id': 'c_bp_group_id',
      'searchKey': 'value',
      'name': 'name',
      '_identifier': '_identifier',
      '_idx': '_idx'
    },
    createStatement: 'CREATE TABLE IF NOT EXISTS c_bp_group (c_bp_group_id TEXT PRIMARY KEY , value TEXT , name TEXT , _identifier TEXT , _idx NUMERIC)',
    dropStatement: 'DROP TABLE IF EXISTS c_bp_group',
    insertStatement: 'INSERT INTO c_bp_group (c_bp_group_id, value, name, _identifier, _idx)  VALUES (?, ?, ?, ?, ?)'
  });

  var BPCategoryList = Backbone.Collection.extend({
    model: BPCategory
  });

  window.OB = window.OB || {};
  window.OB.Model = window.OB.Model || {};
  window.OB.Collection = window.OB.Collection || {};

  window.OB.Model.BPCategory = BPCategory;
  window.OB.Collection.BPCategoryList = BPCategoryList;
}());